app/models/survey_notification.rb

```diff
@@ -39,4 +40,24 @@ def user
   def course
     Course.find(course_id)
   end
+
+  private
+
+  MAX_FOLLOW_UPS = 3
+  def ready_for_follow_up?
+    return false if email_sent_at.nil?
+    return false if Time.now < last_email_sent_at + time_between_emails
+    return false if follow_up_count >= MAX_FOLLOW_UPS
+    true
+  end
+
+  def time_between_emails
+    survey_assignment.follow_up_days_after_first_notification.days
+  end
+
+  def last_email_sent_at
+    sent_at = follow_up_sent_at
+    sent_at ||= email_sent_at
+    sent_at
```

Comment:
It's pretty verbose. Why not `follow_up_sent_at.present? ? follow_up_sent_at : email_sent_at`?